app/src/test/java/org/mozilla/fenix/tabstray/TabsTrayMiddlewareTest.kt

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.tabstray.browser.TabGroup
+import org.mozilla.fenix.telemetry.TabsTrayMiddleware
+import org.mozilla.fenix.utils.Settings
+
+class TabsTrayMiddlewareTest {
```

Comment:
I didn't make tests for the negative (unhappy?) paths of some of these since I couldn't find a way to mock / use any() with the Event classes since they're sealed.

Reply:
This is fine.